Add pid to HostStats

ddn0 opened the pull request:
Along the way fix some clang-tidy nits:
- No default parameters in virtual functions
- Use string literals
- Move over copy
- string::find(char) over string::find(string)


Closes https://github.com/KatanaGraph/katana-enterprise/pull/1574 

The other bits of https://github.com/KatanaGraph/katana-enterprise/pull/1574 have already been implemented separately.

witchel commented on libtsuba/tests/bench_utils.h:
```diff
@@ -26,19 +27,60 @@ inline struct timespec timespec_sub(struct timespec time,
 
 inline int64_t timespec_to_us(struct timespec ts) {
-  return ts.tv_sec * 1'000'000 + ts.tv_nsec / 1'000;
+  return ts.tv_sec * 1'000'000 + ts.tv_nsec / 1'000; // '
```
I really like the ability to add ' into numeric constants.  But my version of cc-mode in emacs likes to count single quote characters.  If there are an odd number, it worries that I haven't closed a character constant.  That comment prevents my emacs from coloring the rest of the buffer as a string.

tylershunt replied:
word. makes sense.

witchel commented:
> Seems fine. Other parts of the code base are using [google benchmark](https://github.com/google/benchmark) I don't think its worth porting there atm, just putting it on the proverbial radar.

Nice.  I looked at it a little when I started this, but it seemed like more than I wanted to take on at the time.  Maybe we can start filtering it in.  Splitting off tsuba_bench makes it a softer target.

I know that real programmers use `llvm::cl::ParseCommandLineOptions`, but I just haven't taken the time yet.

tylershunt commented:
> real programmers use `llvm::cl::ParseCommandLineOptions`

* real programmers that want to keep llvm as a dependency use `llvm::cl::ParseCommandLineOptions`

ddn0 committed:
Add pid to HostStats

Along the way fix some clang-tidy nits:
- No default parameters in virtual functions
- Use string literals
- Move over copy
- string::find(char) over string::find(string)

arekay reviewed (approved):
LGTM.
Might want to wait on @patrickkenney9801 to make sure reports are not impacted.
